shared/validate/validate.go

```diff
@@ -802,3 +802,57 @@ func ParseNetworkVLANRange(vlan string) (int, int, error) {
 
 	return vlanRangeStart, vlanRangeEnd - vlanRangeStart + 1, nil
 }
+
+// IsHostname checks the string is valid DNS hostname.
+func IsHostname(name string) error {
+	err := isName(name)
+	if err != nil {
+		return err
+	}
+
+	match, _ := regexp.MatchString("^[-a-zA-Z0-9]*$", name)
+	if !match {
+		return fmt.Errorf("Name can only contain alphanumeric and hyphen characters")
+	}
+
+	return nil
+}
+
+// IsDeviceName performs the same checks as ValidHostname but also allows underscores.
+func IsDeviceName(name string) error {
+	err := isName(name)
+	if err != nil {
+		return err
+	}
+
+	match, _ := regexp.MatchString("^[-_a-zA-Z0-9]*$", name)
+	if !match {
+		return fmt.Errorf("Name can only contain alphanumeric and hyphen characters")
+	}
+
+	return nil
+}
+
+// isName checks that the given string is 1-63 characters long and does not start or end with a hyphen.
```

Comment:
Should be:
```
// isName checks that the given string is 1-63 characters long, does not start or end with a hyphen and is not a number.
```

Reply:
Although looking at the (copied) code, the error `return fmt.Errorf("Name must not be a number")` seems wrong as its checking that the name doesn't start with a number, not that it isn't a number. So I think both the description and error needing clarifying.

Reply:
Done.